powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java

```diff
@@ -97,7 +100,12 @@ public PushMessageSendResult sendPushMessage(final Long appId, List<PushMessage>
 
             // Iterate over all devices for given user
             for (final PushDeviceRegistrationEntity device : devices) {
-                final PushMessageEntity pushMessageObject = pushMessageDAO.storePushMessageObject(pushMessage.getBody(), pushMessage.getAttributes(), pushMessage.getUserId(), pushMessage.getActivationId(), device.getId());
+                final PushMessageEntity pushMessageObject;
+                if (configuration.isMessageStorageEnabled()) {
```

Comment:
Just for the sake of slightly better code clarity, I would probably isolate this to a private method as well, right above that `updateStatusAndPersist ` method.